phpunit/class-block-templates-test.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * REST API: Block_Templates_Test class
+ *
+ * @package    WordPress
+ * @subpackage REST_API
+ */
+
+/**
+ * Tests for the Block Templates abstraction layer.
+ */
+class Block_Templates_Test extends WP_UnitTestCase {
+	private static $post;
+
+	public static function wpSetUpBeforeClass() {
+		switch_theme( 'tt1-blocks' );
+		gutenberg_register_template_post_type();
+		gutenberg_register_template_part_post_type();
+		gutenberg_register_wp_theme_taxonomy();
```

Comment:
The custom post type and taxo globals are reset between each test so these three functions will still need to be called both here and in `setUp()`. Yup Both.

Reply:
Only if `WP_RUN_CORE_TESTS` is set. There is code in `bootstrap.php` to set it conditionally, but I'm not sure under what circumstances it does get set that way. It doesn't get triggered in my testing.